sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java

```diff
@@ -243,25 +249,33 @@ public URL toURL() throws MalformedURLException {
                 Utility.urlEncode(String.join(",", entry.getValue())));
         }
 
-        return url.toURL();
+        try {
+            return url.toURL();
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsError(new IllegalStateException("The URL parts created a malformed URL."));
+        }
     }
 
     /**
-     * URLParser parses a string URL initializing BlobUrlParts' fields including any SAS-related and snapshot query
-     * parameters. Any other query parameters remain in the UnparsedParams field. This method overwrites all fields
-     * in the BlobUrlParts object.
+     * Parses a string into a BlobUrlParts.
+     *
+     * <p>Query parameters will be parsed into two properties, {@link BlobServiceSasQueryParameters} which contains
+     * all SAS token related values and {@link #getUnparsedParameters() unparsedParameters} which is all other query
+     * parameters.</p>
+     *
+     * <p>If a URL points to a blob in the root container, and the root container is referenced implicitly, i.e. there
+     * is no path element for the container, the name of this blob in the root container will be set as the
+     * containerName field in the resulting {@code BlobURLParts}.</p>
      *
-     * @param url The string URL to be parsed.
-     * @param logger Logger associated to the calling class to log a {@link MalformedURLException}.
+     * @param url The {@code URL} to be parsed.
      * @return A {@link BlobUrlParts} object containing all the components of a BlobURL.
-     * @throws IllegalArgumentException If the {@code url} is malformed.
+     * @throws IllegalArgumentException If {@code url} is a malformed {@link URL}.
      */
-    public static BlobUrlParts parse(String url, ClientLogger logger) {
+    public static BlobUrlParts parse(String url) {
         try {
             return parse(new URL(url));
         } catch (MalformedURLException e) {
-            throw logger.logExceptionAsError(new IllegalArgumentException("Please double check the URL format. URL: "
-                + url));
+            throw new IllegalArgumentException("Please double check the URL format. URL: " + url);
```

Comment:
Error message should be `"Invalid URL format. URL: " + url`

Reply:
Done